### PR TITLE
Update to ili2db4

### LINKED ISSUE #304: Model Baker shows error when used on ili2db4 model
When generating directly from PostGIS I get the following error:

```
Traceback (most recent call last):
  File "C:/Users/KANU/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\QgisModelBaker\gui\generate_project.py", line 285, in accepted
    available_layers = generator.layers()
  File "C:/Users/KANU/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\QgisModelBaker\libqgsprojectgen\generator\generator.py", line 121, in layers
    fields_info = self.get_fields_info(record['tablename'])
  File "C:/Users/KANU/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\QgisModelBaker\libqgsprojectgen\generator\generator.py", line 292, in get_fields_info
    return self._db_connector.get_fields_info(table_name)
  File "C:/Users/KANU/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\QgisModelBaker\libqgsprojectgen\dbconnector\pg_connector.py", line 298, in get_fields_info
    full_name_join=full_name_join))
  File "C:\Programme\OSGeo4W64\apps\Python37\lib\site-packages\psycopg2\extras.py", line 144, in execute
    return super(DictCursor, self).execute(query, vars)
psycopg2.ProgrammingError: column full_name.owner does not exist
LINE 27:                                         ON full_name.owner='...
                                                    ^
```
Reason why I tried to generate the project directly from PostGIS:
I got an error from ili2pg version 3.11.2 that Model Baker is using and therefore had to import model and data separately with the current ili2pg version 4.1.0 beforehand.

### PR DESCRIPTION
**Pull request by m-kuhn**
Open questions:

- [x] **How to deal with existing ili2pg3 schemas**
Affected test: *test_ili2pg_dump_without_metattr (test_dump.TestCustomDump)*
```
Traceback (most recent call last):
  File "/usr/src/QgisModelBaker/tests/test_dump.py", line 45, in test_ili2pg_dump_without_metattr
    available_layers = generator.layers()
  File "/usr/src/QgisModelBaker/libqgsprojectgen/generator/generator.py", line 100, in layers
    fields_info = self.get_fields_info(record['tablename'])
  File "/usr/src/QgisModelBaker/libqgsprojectgen/generator/generator.py", line 268, in get_fields_info
    return self._db_connector.get_fields_info(table_name)
  File "/usr/src/QgisModelBaker/libqgsprojectgen/dbconnector/pg_connector.py", line 303, in get_fields_info
    full_name_join=full_name_join))
  File "/usr/lib/python3/dist-packages/psycopg2/extras.py", line 144, in execute
    return super(DictCursor, self).execute(query, vars)
psycopg2.ProgrammingError: column full_name.colowner does not exist
LINE 27:                                         ON full_name.colowne...
                                                    ^
HINT:  Perhaps you meant to reference the column "full_name.owner".
```
Affected test: **test_export_postgis (test_export.TestExport)**
```
root: INFO: Error: failed to query structure elements Catastro_COL_ES_V2_1_6.Catastro_Registro.miembros

root: INFO: Error:   ERROR: syntax error at or near ")"
  Position: 46
```

- [x] **How to deal with existing ili2gpkg3 schemas**
Affected test: **test_export_geopackage (test_export.TestExport)**
```
root: INFO: failed to read models-table T_ILI2DB_MODEL
  [SQLITE_ERROR] SQL error or missing database (no such column: filename)

```

- [x] **Missing CRS when importing**
Affected test: **test_meta_attr_hidden_toml_postgis (test_projectgen.TestProjectGen)**
```
root: INFO: no CRS for attribute CIAF_LADM.Catastro.Predio.Geometria
```

Fix #304

### CODE REVIEW

**Comment by m-kuhn** [comment]
Does anyone have an idea what could be going wrong with the test that complains about "no CRS for attribute"? ( @yesidpol @gacarrillor )

For everything else, I start to tend towards adding an option to export from ili2db3, so people can at least painlessly update to ili2db4 by exporting/reimporting.

Example workflow:

* Someone tries to "generate" a project from a ili2db3 schema.
* Model baker detects this (e.g. based on the existence of `t_ili2db_attrname.owner`)
* It asks the user to export the data to xtf and reimport it to a newly generated schema (the last part will be up for the user to do himself)

**Comment by pramisters** [comment]
Does the attribute CIAF_LADM.Catastro.Predio.Geometria have a CRS ?

**Comment by m-kuhn** [comment]
@pramisters what does "have a CRS" mean in this context? I probably missed some important updates for ili2db4 (?), the model worked before but now longer works now.

**Comment by gacarrillor** [comment]
> Does anyone have an idea what could be going wrong with the test that complains about "no CRS for attribute"?

Fixed in 745d924

**Comment by gacarrillor** [comment]
As for the open questions (How to deal with existing ili2db3 schemas), from the user perspective it would be great to have an option to "migrate" ili2db3 schemas to ili2db4. We could offer a dialog to "migrate" to ili2db4, involving (probably under the hood):

  + export the ili2db3 schema to XTF (valid for ili2db4) by using --export3; 
  + import the XTF valid for ili2db4 into a new schema (first schema import, then import data).

The dialog would only appear if the user wants to work on an ili2db3 schema and the only input could be an schema name (or db name if using GeoPackage).

File: QgisModelBaker/libqgsprojectgen/db_factory/gpkg_factory.py [hunk 1]
```diff
@@ -49,7 +49,7 @@ def get_tool_version(self):
 
         :return: str ili2gpkg version.
         """
-        return '4.2.0'
+        return '4.3.2'
```
Comment (m-kuhn) [inline QgisModelBaker/libqgsprojectgen/db_factory/gpkg_factory.py]: Are we sure we want to continue like this?

**Comment by m-kuhn** [comment]
@signedav this should be closed I guess?

**Pull request merged**